pytools/persistent_dict.py

```diff
@@ -459,9 +459,13 @@ def __init__(self, identifier: str,
         self.container_dir = container_dir
         self._make_container_dir()
 
+        from pytools import is_sqlite3_fully_threadsafe
+
         # isolation_level=None: enable autocommit mode
         # https://www.sqlite.org/lang_transaction.html#implicit_versus_explicit_transactions
-        self.conn = sqlite3.connect(self.filename, isolation_level=None)
+        self.conn = sqlite3.connect(self.filename,
```

Comment:
Maybe error (or warn?) in the unsafe case?

Reply:
Added a warning in f2f493e

Reply:
Not necessary anymore, the mutex is always used.

Reply:
> Not necessary anymore, the mutex is always used.

Could you explain what you mean here? I'm not sure I understand.

Reply:
I meant that in https://github.com/inducer/pytools/pull/234/commits/4ca16cce520ec36497b3616718d4518f30851f3f, I removed the distinction between safe and unsafe cases, and ust using the mutex always.